src/snowflake/connector/cursor.py

```diff
@@ -882,6 +883,24 @@ def query_result(self, qid):
             )
         return self
 
+    def fetch_arrow_batches(self):
+        self.check_can_use_pandas()
+        if self._query_result_format != "arrow":  # TODO: or pandas isn't imported
```

Comment:
I gave this more thought. I changed the `check_can_use_pandas` to use the `installed_pandas` options.
This way, it enforces that both pyarrow and pandas is installed. Before, we only checked if `pyarrow` was installed. The `fetch_pandas_batches` and `fetch_pandas_all` use this new `check_can_use_pandas`.

The reason why I made this change is because:
- It uses the .options module that we have, thus using one consistent way of checking dependencies.
- Checking that both pandas and pyarrow is installed "enforces" that the user has followed the docs
- The user should have pandas anyway, if they are using the `fetch_pandas*` functions.